[dtensor] refactor get_coordiniate

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #95655
* #95458
* __->__ #95457
* #95456

This refactor get_coordinate to return a optional[list] instead of
directly the coordinate on dim, this is so that we can check if the
rank is inside the mesh easily

Differential Revision: [D43643579](https://our.internmc.facebook.com/intern/diff/D43643579)

wanchaol committed:
[dtensor] refactor get_coordiniate

This refactor get_coordinate to return a optional[list] instead of
directly the coordinate on dim, this is so that we can check if the
rank is inside the mesh easily

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor] refactor get_coordiniate"


This refactor get_coordinate to return a optional[list] instead of
directly the coordinate on dim, this is so that we can check if the
rank is inside the mesh easily

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor] refactor get_coordiniate"


This refactor get_coordinate to return a optional[list] instead of
directly the coordinate on dim, this is so that we can check if the
rank is inside the mesh easily

[ghstack-poisoned]

XilunWu reviewed (approved):
util function signature change. LGTM

wanchaol committed:
Update on "[dtensor] refactor get_coordiniate"


This refactor get_coordinate to return a optional[list] instead of
directly the coordinate on dim, this is so that we can check if the
rank is inside the mesh easily

[ghstack-poisoned]

wanchaol commented:
@wanchaol has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D43643579).
